crates/ruff_python_parser/src/soft_keywords.rs

```diff
@@ -93,7 +98,10 @@ where
                 // 2. The type token is immediately followed by a name token.
                 // 3. The name token is eventually followed by an equality token.
                 Tok::Type => {
-                    if self.start_of_line {
+                    if matches!(
```

Comment:
Unrelated to your change: It's somewhat unfortunate that it's necessary to track whether we're at the start of the line, because we already know this inside of the `Lexer`.

Reply:
Definitely. Perhaps we could move this into the lexer...